modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/parser/hdbtablefunction/XSKHDBTableFunctionParser.java

```diff
@@ -11,33 +11,122 @@
+      inputStream = CharStreams.fromStream(is);
+    }
+
+    HanaLexer lexer = new HanaLexer(inputStream);
+    HanaErrorListener lexerErrorListener = new HanaErrorListener();
+    lexer.removeErrorListeners(); // remove the ConsoleErrorListener
+    lexer.addErrorListener(lexerErrorListener);
+    CommonTokenStream tokenStream = new CommonTokenStream(lexer);
+
+    HanaErrorListener parserErrorListener = new HanaErrorListener();
+    HanaParser parser = new HanaParser(tokenStream);
+    parser.setBuildParseTree(true);
+    parser.removeErrorListeners();
+    parser.addErrorListener(parserErrorListener);
+
+    ParseTree parseTree = parser.sql_script();
+
+//    XSKCommonsUtils.logParserErrors(parserErrorListener.getErrors(),
+//        XSKCommonsConstants.PARSER_ERROR, location,
+//        XSKCommonsConstants.HDB_TABLE_FUNCTION_PARSER);
+//
+//    XSKCommonsUtils.logParserErrors(lexerErrorListener.getErrors(),
+//        XSKCommonsConstants.LEXER_ERROR, location,
+//        XSKCommonsConstants.HDB_TABLE_FUNCTION_PARSER);
+
+    HanaTableFunctionListener listener = new HanaTableFunctionListener();
+    ParseTreeWalker parseTreeWalker = new ParseTreeWalker();
+    parseTreeWalker.walk(listener, parseTree);
+
+    TableFunctionDefinitionModel antlr4Model = listener.getModel();
+    validateAntlrModel(antlr4Model, location);
+
+    return createModel(antlr4Model, parametersModel);
+  }
+
+  private XSKDataStructureHDBTableFunctionModel createModel(TableFunctionDefinitionModel antlrModel,
+      XSKDataStructureParametersModel params) {
+
+    XSKDataStructureHDBTableFunctionModel model = new XSKDataStructureHDBTableFunctionModel();
+    model.setSchema(antlrModel.getSchema());
+    model.setName(antlrModel.getName());
+    model.setLocation(params.getLocation());
+    model.setType(getType());
+    model.setHash(DigestUtils.md5Hex(params.getContent()));
+    model.setCreatedBy(UserFacade.getName());
+    model.setCreatedAt(new Timestamp(new java.util.Date().getTime()));
+    model.setContent(params.getContent());
+    model.setRawContent(params.getContent());
+
+    return model;
+  }
+
+  private void validateAntlrModel(TableFunctionDefinitionModel antlrModel, String location) throws XSKDataStructuresException {
+    try {
+      antlrModel.checkForAllMandatoryFieldsPresence();
+    } catch (Exception e) {
+      XSKCommonsUtils.logCustomErrors(location,
+          XSKCommonsConstants.PARSER_ERROR,
+          "",
+          "",
+          e.getMessage(),
+          XSKCommonsConstants.EXPECTED_FIELDS,
+          XSKCommonsConstants.HDB_VIEW_PARSER,
+          XSKCommonsConstants.MODULE_PARSERS,
+          XSKCommonsConstants.SOURCE_PUBLISH_REQUEST,
+          XSKCommonsConstants.PROGRAM_XSK);
+
+      dataStructuresSynchronizer.applyArtefactState(XSKCommonsUtils.getRepositoryBaseObjectName(location),
+          location, TABLE_FUNCTION_ARTEFACT,
+          ArtefactState.FAILED_CREATE,
+          e.getMessage());
```

Comment:
it will be better if logging in problems view and setting artefact state is handled in central place, based on exception. May be in XSKHDBCoreSynchronizationFacade.parseDataStructureModel but this is for another PR